src/game.cpp

```diff
@@ -11016,6 +11016,11 @@ void game::plfire( bool burst, const tripoint &default_target )
             u.moves -= gun.reload_time(u);
             refresh_all();
         }
+        
+        if( gun.has_flag("FIRE_TWOHAND") && ( !p.has_two_arms() || p.worn_with_flag("RESTRICT_HANDS") ) ) {
+            add_msg(m_info, _("You need two free hands to fire your %s."));
```

Comment:
If you include a format specifier (`%s`) in the format string, you have to add another argument for it, e.g. like this:
```C++
add_msg( _( "The items name is %s." ), gun.tname().c_str() );
```